crystal interpreter cannot find librt

## Bug Report

`crystal i` compiled from master fails on a Fedora 36 toolbox with:

```
cannot find -lrt (/usr/lib64/crystal/librt.so: cannot open shared object file: No such file or directory)
Linker arguments: -L/usr/lib64/crystal -levent -lrt -ldl
Search path: /usr/lib64/llvm13/lib:/usr/lib64/pipewire-0.3/jack/:/lib64:/usr/lib64:/lib:/usr/lib:/usr/lib64/crystal (Crystal::Loader::LoadError)
```

Similar to #11792 but a different library, which is actually present on the system:

```
[sph@toolbox crystal]$ find /usr/lib64 -name "librt.*"
/usr/lib64/librt.so.1
/usr/lib64/librt.a
```

--- a/src/crystal/system/unix/lib_event2.cr
+++ b/src/crystal/system/unix/lib_event2.cr
@@ -1,8 +1,9 @@
 require "c/netdb"
 
-# MUSL: On musl systems, librt is empty. The entire library is already included in libc.
-# The empty library is only available for POSIX compatibility. We don't need to link it.
-{% if flag?(:linux) && !flag?(:musl) %}
+# On musl systems, librt is empty. The entire library is already included in libc.
+# On gnu systems, it's been integrated into `glibc` since 2.34 and it's not available
+# as a shared library.
+{% if flag?(:linux) && flag?(:gnu) && !flag?(:interpreted) %}
   @[Link("rt")]
 {% end %}
 

--- a/src/lib_c/aarch64-linux-gnu/c/dlfcn.cr
+++ b/src/lib_c/aarch64-linux-gnu/c/dlfcn.cr
@@ -1,4 +1,7 @@
-@[Link("dl")]
+{% unless flag?(:interpreted) %}
+  @[Link("dl")]
+{% end %}
+
 lib LibC
   RTLD_LAZY    = 0x00001
   RTLD_NOW     = 0x00002

--- a/src/lib_c/arm-linux-gnueabihf/c/dlfcn.cr
+++ b/src/lib_c/arm-linux-gnueabihf/c/dlfcn.cr
@@ -1,4 +1,7 @@
-@[Link("dl")]
+{% unless flag?(:interpreted) %}
+  @[Link("dl")]
+{% end %}
+
 lib LibC
   RTLD_LAZY    = 0x00001
   RTLD_NOW     = 0x00002

--- a/src/lib_c/i386-linux-gnu/c/dlfcn.cr
+++ b/src/lib_c/i386-linux-gnu/c/dlfcn.cr
@@ -1,4 +1,7 @@
-@[Link("dl")]
+{% unless flag?(:interpreted) %}
+  @[Link("dl")]
+{% end %}
+
 lib LibC
   RTLD_LAZY    = 0x00001
   RTLD_NOW     = 0x00002

--- a/src/lib_c/x86_64-linux-gnu/c/dlfcn.cr
+++ b/src/lib_c/x86_64-linux-gnu/c/dlfcn.cr
@@ -1,4 +1,7 @@
-@[Link("dl")]
+{% unless flag?(:interpreted) %}
+  @[Link("dl")]
+{% end %}
+
 lib LibC
   RTLD_LAZY    = 0x00001
   RTLD_NOW     = 0x00002

